internal/cache/file/downloader/job.go

```diff
@@ -410,6 +424,14 @@ func (job *Job) downloadObjectAsync() {
 		return
 	}
 
+	// Truncate as the parallel downloads can create file of size little higher.
+	err = cacheFile.Truncate(int64(job.object.Size))
```

Comment:
because of the explicit buffer size and buffer is not completely filled? If yes, can you add that info to the comment here.